### PR TITLE
Feature/pde operator backend

### PR DESCRIPTION
**Pull request by samuelburbulla**
Backend changes of #1311.

### CODE REVIEW

**Comment by lululxvi** [comment]
Have you tested if those PI-DeepONet examples can work well?

**Comment by samuelburbulla** [comment]
> Have you tested if those PI-DeepONet examples can work well?
> 
> If so, then update the supported backends (the first line of each file)
> 
> https://github.com/lululxvi/deepxde/blob/b16ac9f9c02be6fc9527930fc276e2fb3dbd905b/examples/operator/advection_aligned_pideeponet.py#L1

Tested all examples, looks good!

With the changes in this PR the aligned examples also run with the `tensorflow` backend.
For the unaligned ones, the backend is missing the `DeepONet` class.

File: deepxde/model.py [hunk 1]
```diff
@@ -281,7 +281,8 @@ def outputs(training, inputs):
             return self.net(inputs)
 
         def outputs_losses(training, inputs, targets, auxiliary_vars, losses_fn):
-            self.net.auxiliary_vars = auxiliary_vars
+            if auxiliary_vars is not None:
```
Comment (lululxvi) [inline deepxde/model.py:284]:
```
self.net.auxiliary_vars = None
if auxiliary_vars is not None:
    self.net.auxiliary_vars = torch.as_tensor(auxiliary_vars)
```

File: deepxde/data/pde_operator.py [hunk 2]
```diff
@@ -262,7 +262,8 @@ def _losses(self, outputs, loss_fn, inputs, model, num_func):
             losses.append(losses_i)
 
         losses = zip(*losses)
-        losses = [bkd.reduce_mean(bkd.as_tensor(l)) for l in losses]
+        # Use stack instead of as_tensor to keep the gradients.
+        losses = [bkd.reduce_mean(bkd.stack(loss, axis=0)) for loss in losses]
```
Comment (lululxvi) [inline deepxde/data/pde_operator.py:266]:
```
losses = [bkd.reduce_mean(bkd.stack(loss, 0)) for loss in losses]
```

**Pull request merged**